master/buildbot/status/buildstep.py

```diff
@@ -267,6 +271,10 @@ def addURL(self, name, url, results=None):
         if results is not None:
             self.urls[name] = {'url': url, 'results': results}
 
+    def addArtifactURL(self, name, url, results=None):
+        self.artifactUrls[name] = url
+        if results is not None:
+            self.artifactUrls[name] = {'url': url, 'results': results}
```

Comment:
Could fix this for self.urls we could probably put duplicated code in a function that is reuse by both